### PR TITLE
[language] Clean up genesis generation and move it under language/tools

### PR DESCRIPTION
**Pull request by dariorussi**
This moves the genesis generation into `language/tools` which is a better home for the tool.
It also cleans it up by refactoring the code and removing the usage of `TransactionExecutor`. 
`TransactionExecutor` is removed and usage of it are using the `VMRuntime` directly. That is
in line with the work that @runtian-zhou was planning to do.
I verified the genesis generated is identical to what it used to be.
`vm-genesis/src/lib.rs` shows as added and removed which is fine as the one re-worked is 
better read with no diff in place and it should be very straightforward

### CODE REVIEW

File: language/tools/test-generation/src/lib.rs [hunk 1]
```diff
@@ -104,9 +102,17 @@ fn execute_function_in_module(
         let mut context =
             TransactionExecutionContext::new(*MAXIMUM_NUMBER_OF_GAS_UNITS, &data_cache);
         let gas_schedule = runtime.load_gas_schedule(&mut context, &data_cache)?;
-        let mut txn_executor =
-            TransactionExecutor::new(&gas_schedule, &data_cache, TransactionMetadata::default());
-        txn_executor.execute_function(&runtime, &module_id, &entry_name, args)
+        let txn_data = TransactionMetadata::default();
```
Comment (dariorussi) [inline language/tools/test-generation/src/lib.rs]: @tzakian take a look at this though I think it should be straightforward

File: language/vm/vm-runtime/src/libra_vm.rs [hunk 2]
```diff
@@ -592,6 +591,18 @@ pub fn is_allowed_script(publishing_option: &VMPublishingOption, program: &[u8])
     }
 }
 
+/// Convert the transaction arguments into move values.
```
Comment (dariorussi) [inline language/vm/vm-runtime/src/libra_vm.rs]: @runtian-zhou I moved this here given it is the only place where it is used. It should be fine and I'll let you re-organize in the future as you see fit

File: language/vm/vm-runtime/src/system_txn/block_metadata_processor.rs [hunk 3]
```diff
@@ -36,27 +39,57 @@ pub(crate) fn process_block_metadata(
...
     result
-        .and_then(|_| txn_executor.make_write_set(Ok(())))
+        .and_then(|_| make_write_set(&mut interpreter_context, &txn_data))
         .and_then(|output| {
             data_cache.push_write_set(output.write_set());
             Ok(output)
         })
 }
+
+fn make_write_set(
```
Comment (dariorussi) [inline language/vm/vm-runtime/src/system_txn/block_metadata_processor.rs]: @runtian-zhou I have moved this function from `TransactionExecutor` here.
I am not sure this is the best place but I'll leave it to you to find a better home given you are actively working in this area

File: language/tools/test-generation/src/lib.rs (hunk 1 above)
Comment (tzakian) [reply language/tools/test-generation/src/lib.rs]: LGTM!

File: language/tools/vm-genesis/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+
+    // create the transaction fee account
+    let transaction_fee_address = account_config::transaction_fee_address();
+    runtime
+        .create_account(
+            interpreter_context,
+            &txn_data,
+            &gas_schedule,
+            transaction_fee_address,
+        )
```
Comment (runtian-zhou) [inline language/tools/vm-genesis/src/lib.rs]: This should be equivalent to expect?

Comment (dariorussi) [reply language/tools/vm-genesis/src/lib.rs]: not sure what you mean here and I assume you are referring to `unwrap_or_else`?
If so yes it's equivalent if not for the fact that by using `format!` clippy complains that we should use `unwrap_or_else`... go figure....

**Comment by dariorussi** [comment]
> LGTM! A question I have is if we want to use VMRuntime or MoveVM as a public API. It seems to me that we should use MoveVM as it would make the initialization phase cleaner but create account might be pretty painful. Would like to hear your thoughts

I think you have a very very good point! I'll commit this as is and review and try to move things into moveVM!
thanks

**Pull request closed without merging**